crates/hir-def/src/nameres/collector.rs

```diff
@@ -1058,8 +1059,37 @@ impl DefCollector<'_> {
         vis: Visibility,
         def_import_type: Option<ImportType>,
     ) -> bool {
-        if let Some((_, v, _)) = defs.types.as_mut() {
-            *v = v.min(vis, &self.def_map).unwrap_or(vis);
+        // `extern crate crate_name` things can be re-exported as `pub use crate_name`.
+        // But they cannot be re-exported as `pub use self::crate_name`, `pub use crate::crate_name`
+        // or `pub use ::crate_name`.
+        //
+        // This has been historically allowed, but may be not allowed in future
+        // https://github.com/rust-lang/rust/issues/127909
+        if let Some((_, v, it)) = defs.types.as_mut() {
+            let is_extern_crate_reimport_without_prefix = || {
+                let Some(ImportOrExternCrate::ExternCrate(_)) = it else {
+                    return false;
+                };
+                let Some(ImportType::Import(id)) = def_import_type else {
+                    return false;
+                };
+                let source = id.import.child_source(self.db);
```

Comment:
We can't use that here, this draws a dependency edge between the defmap query and the AST causing any AST change to invalidate the def map. We need to derive this info from the item tree somehow

Reply:
Okay, I'll try different approach